components/remote_settings/src/lib.rs

```diff
@@ -0,0 +1,228 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+pub mod error;
+use config::RemoteSettingsConfig;
+pub use error::{RemoteSettingsError, Result};
+mod config;
+use rs_client::{Client, ClientConfig};
+use serde::Deserialize;
+use std::{collections::HashMap, fs::File, io::prelude::Write};
+
+uniffi_macros::include_scaffolding!("remotesettings");
+
+pub struct RemoteSettings {
+    pub config: RemoteSettingsConfig,
+    client: Client,
+}
+
+impl RemoteSettings {
+    pub fn new(config: RemoteSettingsConfig) -> Result<Self> {
+        Ok(RemoteSettings {
+            config: config.clone(),
+            client: Client::new(config.into())?,
+        })
+    }
+
+    pub fn get_records(&self) -> Result<RemoteSettingsResponse> {
+        let resp = self.client.get_records()?;
+        let records = resp.json::<RemoteSettingsRecordsResponse>()?.data;
+        let last_modified = resp.headers.get_as::<u64, _>("etag").unwrap()?;
```

Comment:
You can drop the explicit types (ie, the `::<u64, _>`) here and below as the compiler can work it out given it's being put into a struct member with the same type.

It's probably worth a comment here too about the unwrap so casual readers know it is intentional - something like "If the request succeeded, kinto documents the etag header exists and is the timestamp - we assume that's true via the unwrap" or something?

Reply:
Cool, will do! I am intending to do a more thorough docs pass and clean up things like the unused test now that the API is looking good from the Rust perspectiv